app/views/layouts/_adv_search_body.html.haml

```diff
@@ -79,7 +79,7 @@
             %label.control-label.col-md-5
               = _("Global search:")
             .col-md-6
-              - checked = @edit && @edit[:expression] && @edit[:expression][:selected] && @edit[:expression][:selected][:typ] == "global"
+              - checked = @edit && @edit[:expression] && @edit[:expression][:selected] && @edit[:search_type] == "global"
```

Comment:
I am afraid we need to find different solution.

This would work in some cases. But there is another problem underneath.
 - The `@edit[:expression][:selected][:typ]` contains the type of the edited expression (in case you edit the existing one)
 - The `@edit[:search_type` contains the type of the edited expression or what the user last selected or what the last expression was.

The problem is the `or what the last expression was` part. We need to make sure the @edit[:search_type]` gets nil on all occasions when the users switches context from one search to another.